### ISSUE TITLE
feature request: create composite geometric objects

### ISSUE THREAD

**Issue by virginiaMae** [Doenet/DoenetTools#1911]
Example, a text object, a point object, a rectangle and a triangle all grouped together that can be referred to as one object.  Suggested name` <composite>`

**Comment by dqnykamp** [comment]
Would this simply be an extension of the `<group>` component, where we could specify coordinates for the group, and then, inside the group, specified coordinates would be relative to that reference point?

I could also imagine wanting to have or specify additional properties of the group.

Not sure if we could use `<group>` or we'd have to create a new component.  If we wanted to call it `<composite>`, then we'd have to find a new term for "composite components" like copy, map, etc., to avoid confusion.

**Comment by dqnykamp** [comment]
It seems that `<group>` in its current form won't work.  (It is a composite component, so it isn't seen by renderers.)  We could call the new component `<composition>`.  That's what JSXgraph calls a similar concept.

**Comment by dqnykamp** [comment]
For components inside a composition/group, should they all move together in the graph?  JSXgraph has something called a "group" that I could use to underlie this construction.  Inside a group, when you drag one item, they all move together.

**Comment by virginiaMae** [comment]
Yes, I like the simplification of that idea. Currently, I always create a hidden point and then reference that point in all of the other items in the group in order to force the behavior that they are moving together. `composition` seems like a clear term to use. 
It also seems like it would be handy to be able to either define objects within the group initially using absolute coordinates, or relative coordinates but if this is problematic I would lean more towards relative coordinates, with one absolutely defined "center" or "anchor".

**Comment by dqnykamp** [comment]
Technical summary of initial investigation of this issue:
- jsxgraph composition feature doesn't seem to help as it doesn't let one move object together
- jsxgraph group feature looks promising as you can form a group of points that move together
- however, when moving a point in a group, the drag listen of just the point that initiated the move is invoked.  No drag listeners on other points are called.  And there doesn't seem to be a way to bind a listener on the group itself.
- since core needs to know that all points in a group have moved, not just one, we need that fact communicated to core
- one option could be to see if jsxgraph could add a feature for broadcasting that a group has been moved
- another option is to try to use a Context within the React renders to create listeners on items in the group
- a completely different option would be to implement the composition directly in the Doenet core and have the renderers behave like ungrouped objects that get their location instructions from core.
